### PR TITLE
Task for project or layer create/update #8443

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 1]
```diff
@@ -39,6 +39,10 @@
 
     private final ContentService contentService;
 
+    public Context sourceContext;
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:42]:
```suggestion
```

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 2]
```diff
@@ -39,6 +39,10 @@
...
+
+    private Context targetContext;
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:44]:
```suggestion
```

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/SyncContentServiceImpl.java [hunk 3]
```diff
@@ -73,4 +74,12 @@ public void resetInheritance( final ResetContentInheritParams params )
             build().
             execute();
     }
+
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/SyncContentServiceImpl.java:77]:
```suggestion
@Override
```

File: modules/core/core-api/src/main/java/com/enonic/xp/content/ProjectSyncParams.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package com.enonic.xp.content;
+
+import com.google.common.base.Preconditions;
+
+import com.enonic.xp.project.ProjectName;
+
+public final class ProjectSyncParams
+{
+    private final ProjectName sourceProject;
```
Comment (rymsha) [inline modules/core/core-api/src/main/java/com/enonic/xp/content/ProjectSyncParams.java:9]: I am not sure source project should be here.
1. It allows to sync content from any project
2. It does not fit into muli-inheritance 

Service should get sourceProject(s) from project data

`targetProject` should be called just `project`

File: modules/itest/itest-core/src/test/java/com/enonic/xp/core/content/ProjectEventListenerTest.java [hunk 5]
```diff
@@ -71,7 +70,7 @@ public void testChildren()
         compareSynched( sourceChild2, targetChild2 );
     }
 
-    @Test
+   /* @Test
```
Comment (rymsha) [inline modules/itest/itest-core/src/test/java/com/enonic/xp/core/content/ProjectEventListenerTest.java:73]: remove

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/task/ProjectsSyncTask.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    private void doSync( final Project sourceProject, final Project targetProject, final ProgressReporter progressReporter )
+    {
+        syncContentService.syncProject( ProjectSyncParams.create().
+            sourceProject( sourceProject.getName() ).
+            targetProject( targetProject.getName() ).
+            build() );
+    }
+
+    private Context createAdminContext()
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/task/ProjectsSyncTask.java:72]: Maybe transfer task caller context, so it is not always "su"?

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/itest/itest-core/build.gradle [hunk 7]
```diff
@@ -11,4 +11,8 @@ dependencies {
     testImplementation( testFixtures( project(":core:core-blobstore") ) )
     testImplementation( testFixtures( project(":core:core-repo") ) )
     testImplementation( testFixtures( project(":core:core-security") ) )
+    testImplementation( testFixtures( project(":core:core-task") ) )
+    testImplementation( testFixtures( project(":core:core-internal") ) )
+
+    testFixturesImplementation(testFixtures( project( ":core:core-task" ) ))
```
Comment (rymsha) [inline modules/itest/itest-core/build.gradle:17]:
```suggestion
    testImplementation(testFixtures( project( ":core:core-task" ) ))
```

File: modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/ProjectResource.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    implements JaxRsComponent
+{
+    private ProjectService projectService;
+
+    private SyncContentService syncContentService;
+
+    private TaskService taskService;
+
+    @POST
+    @Path("syncAll")
```
Comment (rymsha) [inline modules/server/server-rest/src/main/java/com/enonic/xp/impl/server/rest/ProjectResource.java:37]: Write a doc

**Review by rymsha — APPROVED** [review]

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**